ts/features/idpay/code/onboarding/screens/IdPayCodeDisplayScreen.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+  Banner,
+  Body,
+  ButtonSolid,
+  H2,
+  IOColors,
+  IOSpacingScale,
+  Link,
+  VSpacer
+} from "@pagopa/io-app-design-system";
+import * as React from "react";
+import { StyleSheet, View } from "react-native";
+
+import { useNavigation } from "@react-navigation/native";
+import { SafeAreaView } from "react-native-safe-area-context";
+import { NewH3 } from "../../../../../components/core/typography/NewH3";
+import TopScreenComponent from "../../../../../components/screens/TopScreenComponent";
+import I18n from "../../../../../i18n";
+import { emptyContextualHelp } from "../../../../../utils/emptyContextualHelp";
+
+const CodeDigitDisplayBox = ({ digit }: { digit: string }) => (
+  <View style={styles.codeDigit}>
+    <NewH3>{digit}</NewH3>
+  </View>
+);
+const CodeDisplayComponent = ({ code }: { code: string }) => (
+  <View style={styles.codeDisplay}>
+    {code.split("").map((digit, index) => (
+      <CodeDigitDisplayBox key={index} digit={digit} />
+    ))}
+  </View>
+);
+
+export const IdPayCodeDisplayScreen = () => {
+  const code = "12345"; // get from store
+  const viewRef = React.useRef(null);
+  const navigation = useNavigation(); // TODO:: delete
+  return (
+    <SafeAreaView style={styles.mainContainer}>
+      <TopScreenComponent contextualHelp={emptyContextualHelp}>
+        <H2>{I18n.t("idpay.code.onboarding.header")}</H2>
+        <VSpacer size={16} />
+        <Body color="grey-700" weight="Regular">
+          {I18n.t("idpay.code.onboarding.body1")}
+        </Body>
+        <Body color="grey-700" weight="Bold">
+          {I18n.t("idpay.code.onboarding.bodyBold")}
+        </Body>
+        <Link>{I18n.t("idpay.code.onboarding.bodyCta")}</Link>
+        <VSpacer size={24} />
+        <CodeDisplayComponent code={code} />
+        <VSpacer size={24} />
+        <Banner
+          color="neutral"
+          pictogramName="help" // security once new DS ver is released
+          size="big"
+          viewRef={viewRef}
+          title={I18n.t("idpay.code.onboarding.banner.header")}
+          content={I18n.t("idpay.code.onboarding.banner.body")}
+        />
+      </TopScreenComponent>
+      <ButtonSolid
+        accessibilityLabel={I18n.t("global.buttons.continue")}
+        label={I18n.t("global.buttons.continue")}
+        fullWidth={true}
+        onPress={() => navigation.goBack()}
+      />
+    </SafeAreaView>
+  );
+};
+
+const styles = StyleSheet.create({
+  mainContainer: {
+    flex: 1,
+    paddingHorizontal: IOSpacingScale[6] // 24
```

Comment:
I don't think it's a robust solution. Imagine if, some day, someone adds another scale value at position 3 or 4. Every scale is then pushed by one position to the right and the padding used here is changed.

```suggestion
    paddingHorizontal: 24
```

Reply:
In this specific case it could be `appMarginDefault` because it makes sense. But in general you should define the spacing variables in the following way:

```ts
const spacingVariable: IOSpacingScale = 24
```

Reply:
addressed in [5568f2b](https://github.com/pagopa/io-app/pull/4992/commits/5568f2bd729774a0710952150867f502b54aa7f3) 👍